client/galaxy/scripts/components/login/Login.vue

```diff
@@ -25,113 +26,136 @@
+            }
         },
-        reset: function(ev) {
-            let rootUrl = getAppRoot();
-            ev.preventDefault();
-            axios
-                .post(`${rootUrl}user/reset_password`, { email: this.login })
-                .then(response => {
-                    this.messageVariant = "info";
-                    this.messageText = response.data.message;
-                })
-                .catch(error => {
-                    this.messageVariant = "danger";
-                    let message = error.response.data && error.response.data.err_msg;
-                    this.messageText = message || "Password reset failed for an unknown reason.";
-                });
+        methods: {
+            toggleLogin: function () {
+                if (this.$root.toggleLogin) {
+                    this.$root.toggleLogin();
+                }
+            },
+            submitGalaxyLogin: function (method) {
+                let rootUrl = getAppRoot();
+                axios
+                    .post(`${rootUrl}user/login`, this.$data)
+                    .then(response => {
+                        if (response.data.message && response.data.status) {
+                            alert(response.data.message);
+                        }
+                        if (response.data.expired_user) {
+                            window.location = `${rootUrl}root/login?expired_user=${response.data.expired_user}`;
+                        } else if (response.data.redirect) {
+                            window.location = encodeURI(response.data.redirect);
+                        } else {
+                            window.location = `${rootUrl}`;
+                        }
+                    })
+                    .catch(error => {
+                        this.messageVariant = "danger";
+                        let message = error.response.data && error.response.data.err_msg;
+                        this.messageText = message || "Login failed for an unknown reason.";
+                    });
+            },
+            submitOIDCLogin: function (method) {
+                let rootUrl = getAppRoot();
+                axios
+                    .post(`${rootUrl}authnz/google/login`)
+                    .then(response => {
+                        if (response.data.redirect_uri) {
+                            window.location = encodeURI(response.data.redirect_uri);
+                        }
+                        // Else do something intelligent or maybe throw an error -- what else does this endpoint possibly return?
+                    })
+                    .catch(error => {
+                        this.messageVariant = "danger";
+                        let message = error.response.data && error.response.data.err_msg;
+                        this.messageText = message || "Login failed for an unknown reason.";
+                    });
+            },
+
+            submitElixirLogin: function (method) {
```

Comment:
Pretty much similar to the `submitOIDCLogin` method, would it be possible to leverage that function and set idp (i.e., Google and Elixir) as a variable that defines which backend to post to? i.e., `authnz/google/login` vs `authnz/elixir/login`

Reply:
@kidanem I guess we can pretty much simplify this UI implementation if we leverage a common method. It can also help extending it with new IdPs.

Reply:
@VJalili - I agree with your suggestion. I have just been waiting for the PSA pull request so we can do it in one go. The PSA pull request is now accepted and our config changes should be in the next PSA release. 
I will look into enhancing this part soon.

Reply:
Yes, I agree all can happen once a newer psa version is release.